front/app/feed.xml/route.tsx

```diff
@@ -39,16 +40,24 @@ export async function GET() {
 
   comicsWithMetadata.forEach((comicWithMetadata) => {
     const { comic, metadata } = comicWithMetadata;
-    let description = "";
-    comic.image_urls.forEach((img: string) => {
-      const image = metadata.images[img];
-      description += `<img 
-        src="${getImageUrl(image)}"
-        alt="${image.ocr}"
-        width="${image.width}"
-        height="${image.height}"
-      />`;
-    });
+    let description = renderToString(
+      <>
+        {comic.image_urls.map((image_url) => {
+          const image = metadata.images[image_url];
+          return (
+            <img
+              src={getImageUrl(image)}
+              alt={image.ocr}
+              width={image.width}
+              height={image.height}
+            />
+          );
+        })}
+        <p>
+          <a href={comic.link}>Original Post</a>
+        </p>
+      </>,
```

Comment:
Could we extract this logic to a separate react component? Something called  RSSdescription in the components folder would be nice.

Reply:
i did try that but it seemed like an unnesaserry complication since the route file would have to be renamed anyway to use the component. happy to switch to that though if you think it makes sense

Reply:
@TomW1605 I meant moving this JSX code (The `<p>bracket things</p>`) to a separate file and then passing some `<RSSdescription metadata={metadata}/>` to the `renderToString` in the `route.tsx`.

This way we will have one place with all the RSS logic and another with _the html for description_. When changing the description we don't need to worry about all the RSS feed things (e.g. that we render it to string etc.), and when we change the RSS logic we don't need to worry about how the description looks like.

Reply:
ok, yeah, that works. i need to pass through the comic and the images not just the metadata otherwise we end up redoing all the sorting process as well but we end up with this in the route file
```js
let description = renderToString(
  <>
    <RSSdescription comic={comic} images={metadata.images} />
  </>,
);
```